ecosystem/indexer-grpc/indexer-grpc-table-info/src/table_info_parser.rs

```diff
@@ -93,7 +92,23 @@ impl TableInfoParser {
             tasks.push(task);
         }
         match futures::future::try_join_all(tasks).await {
-            Ok(res) => res,
+            Ok(res) => {
+                let db_reader = db.reader.clone();
+                let last_version = std::cmp::min(
+                    self.highest_known_version,
+                    self.current_version + (self.parser_batch_size * self.parser_task_count) as u64,
+                );
+                let mut retried = 0;
+                while !db_reader.is_indexer_async_v2_pending_on_empty().unwrap() {
+                    retried += 1;
+                    info!("Missing data in table info parsing at start version {:?}, retrying {:?} times", self.current_version, retried);
```

Comment:
This should be structured, i.e. info!(current_version = self.current_version, retry_count = retried, "blabla")